sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -77,4 +78,56 @@ class QueryCompilationErrorsSuite extends QueryTest with SharedSparkSession {
       " but got: org.apache.spark.sql.types.NumericType"))
   }
 
+  test("UNSUPPORTED_GROUPING_EXPRESSION: filter with grouping/grouping_Id expression") {
+    val df = Seq(
+      (536361, "85123A", 2, 17850),
+      (536362, "85123B", 4, 17850),
+      (536363, "86123A", 6, 17851)
+    ).toDF("InvoiceNo", "StockCode", "Quantity", "CustomerID")
+
+    // filter with grouping
+    var errMsg = intercept[AnalysisException] {
+      df.filter("grouping(CustomerId)=17850")
+        .groupBy("CustomerId").agg(Map("Quantity" -> "max"))
+    }
+    assert(errMsg.message ===
+      "grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup")
+    assert(errMsg.errorClass === Some("UNSUPPORTED_GROUPING_EXPRESSION"))
+
+    // filter with grouping_id
+    errMsg = intercept[AnalysisException] {
+      df.filter("grouping_id(CustomerId)=17850").
+        groupBy("CustomerId").agg(Map("Quantity" -> "max"))
+    }
+    assert(errMsg.errorClass === Some("UNSUPPORTED_GROUPING_EXPRESSION"))
+    assert(errMsg.message ===
+      "grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup")
+  }
+
+  test("UNSUPPORTED_GROUPING_EXPRESSION: Sort with grouping/grouping_Id expression") {
+    val df = Seq(
+      (536361, "85123A", 2, 17850),
+      (536362, "85123B", 4, 17850),
+      (536363, "86123A", 6, 17851)
+    ).toDF("InvoiceNo", "StockCode", "Quantity", "CustomerID")
+
+   // Sort with grouping
+    var errMsg = intercept[AnalysisException] {
+      df.sort(grouping("CustomerId"))
+        .groupBy("CustomerId").agg(Map("Quantity" -> "max"))
+    }
+    assert(errMsg.errorClass === Some("UNSUPPORTED_GROUPING_EXPRESSION"))
+    assert(errMsg.message ===
+      "grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup")
```

Comment:
Seems like we can already pass a parametrized reason here as we know this is sort with grouping?

Reply:
If I am understanding correctly So the message should be like this 
"Sort with grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup" instead of 
"grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup"

Similar case for filter
"Filter with grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup" instead of 
"grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup"

Reply:
As per the code `val groupingExprs = findGroupingExprs(child)`  . findGroupingExprs is called from two places

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L733)

 and 

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L740.

In that case if want to parameterised the error by updating the input param to the method 
`def findGroupingExprs(plan: LogicalPlan) ` to  `def findGroupingExprs(plan: LogicalPlan, parentInfo: String)`

So that when throwing the exception we can add the where it failed either in filter or sort.

If this looks fine, I can make that change

Reply:
NVM. `Sort with grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup` is not really more helpful.